src/HPolyhedron.jl

```diff
@@ -499,6 +501,43 @@ function remove_redundant_constraints!(P::PT;
     return P
 end
 
+"""
+    linear_map(M::AbstractMatrix{N}, P::PT) where {N<:Real, PT<:HPoly{N}}
+
+Concrete linear map of a polyhedron in constraint representation.
+
+### Input
+
+- `M` -- matrix
+- `P` -- polyhedron in constraint representation
+
+### Output
+
+A polyhedron of the same type as the input (`PT`).
+
+### Algorithm
+
+If the matrix ``M`` is invertible (which we check with a sufficient condition),
+then ``y = M x`` implies ``x = \\text{inv}(M) y`` and we transform the
+constraint system ``A x ≤ b`` to ``A \\text{inv}(M) y ≤ b``.
+"""
+function linear_map(M::AbstractMatrix{N}, P::PT) where {N<:Real, PT<:HPoly{N}}
+    if !isinvertible_sufficient(M)
+        if P isa HPolyhedron
+            error("linear maps for polyhedra need to be invertible")
+        end
+        # use the implementation for general polytopes
+        return invoke(linear_map, Tuple{typeof(M), AbstractPolytope{N}}, M, P)
+    end
+    # matrix is invertible
+    invM = inv(M)
```

Comment:
`inv` has complexity O(n^3) (see [here](https://en.wikipedia.org/wiki/Computational_complexity_of_mathematical_operations#Matrix_algebra)), see also [this report](http://www.diva-portal.org/smash/get/diva2:1219121/FULLTEXT01.pdf); we could probably be better off with a lazy matrix inverse (related [LazyArrays.jl](https://github.com/JuliaArrays/LazyArrays.jl)). i propose to move this comment to another issue.